Localization, HTTPS, Paste URL, fixes

jsteinaker opened the pull request:
Hey, it's been a while.

Apparently, about a year ago I added preliminary support for localization, fixed some crashes and did some minor rework (I honestly can't remember that much). Today I picked up where I left and now the "Paste Shoutcast URL" it's actually working. If you have the station's stream address in the clipboard it will add it to your list. If the address points to a PLS file, it can recover the station's name from there and use it.

Tried and working here.

pulkomandy commented on Station.cpp:
```diff
@@ -455,22 +480,23 @@ Station::LoadIndirectUrl(BString& sUrl) {
             char* title = regFind(body, patternTitle);
             if (title) station->fName.SetTo(title);
 
+			/*
```
use #if 0 and add a comment explaining why it's disabled.

KapiX commented:
> @KapiX what's the process to add it to the localization tool?

It depends on who maintains the project:
* People who want to host personal projects can message me or @humdingerb to get developer access in the tool. That allows creating a project and uploading catkeys.
* Projects under HaikuArchives umbrella are maintained by @humdingerb (and sometimes me), though help is welcome.

humdingerb commented:
Once this PR is merged, I can give it a quick once-over for sentence casing and then add it to Polyglot.

jsteinaker commented on Station.cpp:
```diff
@@ -171,48 +174,69 @@ status_t Station::RetrieveStreamUrl() {
+	// Creating a socket and trying to connect to the specified IP/port pair is enough for our purposes. That
+	// way, if a server is down we can use the next DNS record, and craft a new URL like http://192.168.0.1/path/to/stream
+	// 
+	// On HTTPS, unfortunately, this isn't possible, as the certificate is associated to the hostname, and so we have to
```
This one is coming from last year. I'll take a look, I can't remember now but it's possible that I've already opened a bug report. Otherwise I'll just do it.

jsteinaker replied:
I've filled a [bug report](https://dev.haiku-os.org/ticket/16155#ticket)

jsteinaker commented on Makefile:
```diff
@@ -25,7 +25,7 @@ PACKAGE = $(TARGET_DIR)/$(NAME)_$(VERSION)-$(ARCH).hpkg
 TYPE = APP
 
 # 	If you plan to use localization, specify the application's MIME signature.
-APP_MIME_SIG = application/x-vnd.Fishpond-Radio
```
Didn't know, sorry.

jsteinaker commented on MainWindow.cpp:
```diff
@@ -154,12 +168,12 @@ void MainWindow::MessageReceived(BMessage* message) {
             if (stationItem) {
 				Station* station = stationItem->GetStation();
-                BString statusText("Probing station %s %s");
+                BString statusText(B_TRANSLATE("Probing station %s %s"));
```
Valid point. I'll take a look.

jsteinaker commented on MainWindow.cpp:
```diff
@@ -140,11 +146,19 @@ void MainWindow::MessageReceived(BMessage* message) {
+                    	fSettings->Stations->Save();
+                	} else {
+					BString msg;
+					msg.SetToFormat("Station %s did not respond correctly and could not be added", 
```
Yep. I added this one yesterday and forgot about it.

jsteinaker commented on MainWindow.cpp:
```diff
@@ -140,11 +146,19 @@ void MainWindow::MessageReceived(BMessage* message) {
-                        fSettings->Stations->Save();
-                    }
+                if (station) {
+                	status_t probeStatus = station->Probe();
```
Indeed, seems that I've fucked up

jsteinaker commented:
@pulkomandy many thanks for your review. I'll put all the suggested changes in a new commit and I'll change the PR to point there, if that's the right way to do it.

jsteinaker commented:
I've added some minor features that were requested on Issues (help menu, ability to switch between one-stream and multiple streams modes) and updated the spanish catkeys. The features discussed previously should be all working now (if you can excuse the code quality coming from a non-programmer). Can you guys give it a try and confirm if everything is working for you as well?

Vidrep commented:
Built and tested on 64 bit. Starting a stream causing a crash. Link to “help” is working ok. If parallel streams are selected, the app freezes up and has to be terminated via TeamMonitor.

Vidrep commented:
tested on 32 bit and features seem to work ok, including parallel playback and  help.

Vidrep commented:
Went back to 64 bit and compiled again. The problem occurs when you try to use a list of stations from a previously existing install. Deleting the contents of the "Stations" directory, and adding the stations again solves the playback issue.

jsteinaker commented:
Thanks for reporting. I've just made a minor bugfix cause I realized the menu wasn't reflecting the parallel playback status on startup (it wasn't being ticked), even when the setting was working OK.

Regarding the crash you're experiencing, I shouldn't have touched anything in the Station class, nor the Stations directory. The config file now stores the parallel playback status, but that's stored on config/settings/StreamRadio.settings

humdingerb commented on StationFinderRadioNetwork.cpp:
```diff
@@ -25,7 +29,7 @@ StationFinderRadioNetwork::StationFinderRadioNetwork()
     fIconLookupList()
 {
-    serviceName.SetTo("Radio Network");
+    serviceName.SetTo(B_TRANSLATE("Radio Network"));
```
This isn't a name, I think, or is it? If not, sentence casing.

```suggestion
    serviceName.SetTo(B_TRANSLATE("Radio network"));
```

jsteinaker replied:
It might have been. On their website it states "Community Radio Browser". I'll use that, it makes more sense.

humdingerb commented on StationFinderListenLive.cpp:
```diff
@@ -216,7 +220,7 @@ StationFinderListenLive::ParseCountryReturn(BMallocIO* data, const char* searchF
 		doc[matches[6].rm_eo] = 0;
 		doc[matches[8].rm_eo] = 0;
-		printf("Station %s at %s in %s address %s, data rate=%s, genre=%s\n", 
+		printf(B_TRANSLATE("Station %s at %s in %s address %s, data rate=%s, genre=%s\n"), 
```
Normally, debug output isn't translated. If you do want to, you may want to use B_TRANSLATE_COMMENT to explain the variables. At least the first 4...
Maybe with an example. I'm guessing: "Station 'station name' at 'URL' in 'country'... hmm, what's the address?

jsteinaker replied:
I'll check this. I never took a look at this service, because it wasn't even working in first place. It might have moved to a different website or maybe it just went down. Anyway, I'll leave the debug output in English and I'll take a closer look at "ListenLive" to see what happened.

Vidrep commented:
humdinger, one of the things we should strive for in HaikuArchives apps is consistency in the menu items. Some apps have a help menu item with a link to a user guide, and the project page. Others have those items listed under "app", some under "file", etc. Some apps are missing these items alltogether.

jsteinaker commented on MainWindow.cpp:
```diff
@@ -30,25 +31,39 @@
+        	.AddItem(menuParallelPlayback)
+            .AddItem(B_TRANSLATE("Paste Shoutcast URL"), MSG_PASTE_URL)
+            .AddItem(B_TRANSLATE("Check Station"), MSG_CHECK)
+            .AddItem(B_TRANSLATE("Remove"), MSG_REMOVE, B_DELETE)
```
I'll change it. Command (which I have set to CTRL) + DEL is working for me, but it's counter-intuitive, specially considering that, as you said, it doesn't have a symbol. I might take a look at that on the Haiku codebase, but it's probably not easy to resolve.

jsteinaker commented:
All done. On the search window, I've noticed that some of the strings are still not translated ("keyword", "country", etc) but I've already done the catkeys at that point. I'll leave those for the next commit, I'll try to get the listenlive.eu service back to work (I think they've moved to radiomap.eu)

humdingerb commented:
Thanks, Javier!

> Some apps have a help menu item with a link to a user guide, and the project page. Others have those items listed under "app", some under "file", etc. Some apps are missing these items alltogether.

Ideally, yes. If an application has no file related menu items, the 'main' menu with Quit, About, Help etc. should be in a menu named "App". Otherwise those are fine under "File", I suppose. That's at least the tradition...

Those would be easy tasks, with plenty of examples around. If the apps afflicted with these deficiencies were identified, people interested in starting to contribute code could work on it. Or we'd make a few tasks of it for next GCI.
